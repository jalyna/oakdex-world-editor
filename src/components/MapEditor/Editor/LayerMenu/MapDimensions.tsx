import * as React from 'react'
import styled from 'styled-components'
import { Dispatch } from 'redux'
import { connect } from 'react-redux'

import { GREY_70, GREY_95 } from 'shared/theme'

interface MapDimensionsProps {
  width: number,
  height: number
}

function mapStateToProps ({ mapData }: any) {
  return {
    width: mapData.width,
    height: mapData.height
  }
}

function MapDimensions ({ width, height }: MapDimensionsProps) {
  return (
    <Wrapper>
      {width} x {height}
    </Wrapper>
  )
}

const Wrapper = styled.div`
  margin-bottom: 5px;
  font-weight: bold;
  border-bottom: 1px solid ${GREY_70};
  background: ${GREY_95};
  padding: 5px;
`

export default connect(
  mapStateToProps
)(MapDimensions)
