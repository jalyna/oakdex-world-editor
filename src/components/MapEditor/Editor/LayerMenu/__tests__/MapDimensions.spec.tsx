import * as React from 'react'
import configureStore, { MockStoreEnhanced } from 'redux-mock-store'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import MapDimensions from '../MapDimensions'

const mockStore = configureStore()
let store = null as MockStoreEnhanced | null

describe('<MapDimensions>', () => {
  beforeEach(() => {
    store = mockStore({
      mapData: {
        width: 12,
        height: 10
      }
    })
  })

  it('renders correctly', () => {
    const editor = mount(<Provider store={store}><MapDimensions /></Provider>)
    expect(toJson(editor, { noKey: false, mode: 'deep' })).toMatchSnapshot()
  })
})
