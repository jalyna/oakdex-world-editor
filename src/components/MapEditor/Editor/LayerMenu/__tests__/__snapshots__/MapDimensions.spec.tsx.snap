// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapDimensions> renders correctly 1`] = `
<Connect(MapDimensions)>
  <styled.div
    onClick={[Function]}
  >
    <div
      className="sc-cLQEGU hhjOvE"
      onClick={[Function]}
    >
      12
       x 
      10
    </div>
  </styled.div>
</Connect(MapDimensions)>
`;
